File: kotlinStudy2024/src/javaPackage/lec04/Lec04Main.java
package javaPackage.lec04;

public class Lec04Main {

  public static void main(String[] args) {
    JavaMoney money1 = new JavaMoney(1_000L);
    JavaMoney money2 = new JavaMoney(2_000L);
    System.out.println(money1.plus(money2));

    if(money1.compareTo(money2) == 0) {
      System.out.println("객체비교에서 자동으로 CompareTo 호출 x");
    }

    if(money1.equals(money2)) {
      System.out.println("동등성 비교에서도 자동으로 Equals 호출 x");
    }

    // 객체를 더할 때도 연산자 사용 못함
    System.out.println(money1.plus(money2));
  }
}

File: kotlinStudy2024/src/kotlinPackage/lec04/Lec04Main.kt
package kotlinPackage.lec04

import javaPackage.lec04.JavaMoney

fun main() {
    val money1 = JavaMoney(3_00L)
    val money2 = JavaMoney(2_00L)

    if(money1 > money2) {
        println("객체비교에서 자동으로 CompareTo 호출")
    }

    if(money1 == money2) {
        println("동등성 비교에서도 자동으로 Equals 호출")
    }

    val money3 = Money(1_000L)
    val money4 = Money(2_000L)

    // 연산자를 사용해도 무방
    println(money3 + money4)
}